pkg/controller/pxc/users.go

```diff
@@ -847,6 +942,81 @@ func (r *ReconcilePerconaXtraDBCluster) updateUserPass(cr *api.PerconaXtraDBClus
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) discardOldPassword(cr *api.PerconaXtraDBCluster, secrets, internalSecrets *corev1.Secret, user *users.SysUser) error {
+	pxcUser := users.Root
+	pxcPass := string(internalSecrets.Data[users.Root])
+	if _, ok := secrets.Data[users.Operator]; ok {
+		pxcUser = users.Operator
+		pxcPass = string(internalSecrets.Data[users.Operator])
+	}
+
+	addr := cr.Name + "-pxc." + cr.Namespace
+	if cr.CompareVersionWith("1.6.0") >= 0 {
+		addr = cr.Name + "-pxc-unready." + cr.Namespace + ":33062"
+	}
+	um, err := users.NewManager(addr, pxcUser, pxcPass, cr.Spec.PXC.ReadinessProbes.TimeoutSeconds)
+	if err != nil {
+		return errors.Wrap(err, "new users manager")
+	}
+	defer um.Close()
+
+	err = um.DiscardOldPassword(user)
+	if err != nil {
+		return errors.Wrap(err, "discard old user pass")
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) isPassPropagated(cr *api.PerconaXtraDBCluster, user *users.SysUser) (bool, error) {
+	components := map[string]int32{
+		"pxc": cr.Spec.PXC.Size,
+	}
+
+	if cr.ProxySQLEnabled() {
+		components["proxysql"] = cr.Spec.ProxySQL.Size
+	} else if cr.HAProxyEnabled() {
+
+		components["haproxy"] = cr.Spec.HAProxy.Size
+	}
+
+	// TODO: wrap checking in a retry block in a way that we retry only
```

Comment:
I don't think we can let it to reconciliation loop, since we already updated the internal secret. The next reconciliation will see passwords are equal in both secrets and return early.

Reply:
Yes, exactly the case I'm working on, but leaving it to reconcile loop is not the issue, when we get an error from checking propagation we will still end up in this situation, we simply now have a case where we somehow need to see if the old pass is discarded.